examples/reshare/src/dkg/actor.rs

```diff
@@ -0,0 +1,138 @@
+use super::{Mailbox, Message};
+use crate::dkg::DkgManager;
+use commonware_cryptography::{
+    bls12381::{
+        dkg::player::Output,
+        primitives::{group::Share, poly::Public, variant::MinSig},
+    },
+    ed25519::{PrivateKey, PublicKey},
+    Sha256,
+};
+use commonware_p2p::{Receiver, Sender};
+use commonware_runtime::{tokio, Handle, Spawner};
+use futures::{channel::mpsc, StreamExt};
+use std::cmp::Ordering;
+use tracing::info;
+
+/// The number of blocks in an epoch.
+const EPOCH_LENGTH: u64 = 150;
+
+pub struct Actor {
+    context: tokio::Context,
+    mailbox: mpsc::Receiver<Message<Sha256, PrivateKey, MinSig>>,
+    signer: PrivateKey,
+    contributors: Vec<PublicKey>,
+}
+
+impl Actor {
+    /// Create a new DKG [Actor] and its associated [Mailbox].
+    pub fn new(
+        context: tokio::Context,
+        signer: PrivateKey,
+        mut contributors: Vec<PublicKey>,
+        mailbox_size: usize,
+    ) -> (Self, Mailbox<Sha256, PrivateKey, MinSig>) {
+        // Sort the list of contributors to ensure everyone agrees on ordering.
+        contributors.sort();
+
+        let (sender, mailbox) = mpsc::channel(mailbox_size);
+        (
+            Self {
+                context,
+                mailbox,
+                signer,
+                contributors,
+            },
+            Mailbox::new(sender),
+        )
+    }
+
+    /// Start the DKG actor.
```

Comment:
I suspect this example could be a lot simpler if you put this logic into `application`?